### ISSUE TITLE
ActivityIdLayoutRenderer  - Wrong class attributes

### ISSUE THREAD

**Issue by snakefoot** [ARKlab/Ark.Tools#6]
Think you have applied the wrong attributes here, as they are very special:

https://github.com/ARKlab/Ark.Tools/blob/573685ce3e101b640350c9ac1248bff01c71813f/Ark.Tools.NLog/ActivityIdLayoutRenderer.cs#L14-L17

- ` [ThreadAgnostic]` - Means that the layoutrenderer will generate the same output independent on which threads it happens (Ex. like the FormattedMessage). But this is not true for [System.Diagnostics.Activity.Id](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.id) since it is only valid for the active thread logging, and not available if writing happens on a background-thread (`<targets async="true">`).

- `[AppDomainFixedOutput]`  - Means that the layoutrenderer will always generate the same output independent of time and thread (Ex. the HOSTNAME). But this is also not true for [System.Diagnostics.Activity.Id](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.id).

I suggest that you change to this instead:

```c#
 [LayoutRenderer("ark.activityid")] 
 [ThreadSafe] 
 public class ActivityIdLayoutRenderer : LayoutRenderer 
```

Also notice that [NLog.DiagnosticSource](https://www.nuget.org/packages/NLog.DiagnosticSource) is available, but that will ofcourse be another dependency, that might be unwanted.

Can see that you have been working on implementing helper-methods to make it easier to setup NLog Targets (File + Console + Mail + Database) from code. NLog 5.0 introduces something similar out-of-the-box. Would be grateful if you could take a peek at the [current prototype](https://github.com/NLog/NLog/pull/4323), and give input whether it matches your expectations.

**Comment by AndreaCuneo** [comment]
Hi @snakefoot, thanks a lot for checking and pointing out the issue!

Indeed probably the ActivityId has been added later and missed to update the class Attributes.
We didn't suffered any issue with them tbh, so the error rate must be very low. 

The added dependency is not a big issue as looks maintained and with a sane License. I can probably migrate to it, and leaving 'ark.activityId', fixed, for a bit. 
As far as you know, would be possible to derive from the NLog.DiagnosticSource LayoutRender and 'alias' the Layout name and the behaviour (default to `Activity.Id` instead of `TraceId`)? 
Not so sure if NLog would complain about Attributes being defined twice over a class across its hierarchy.

Regarding the mentioned prototype, I'm going to take a look and give my feedbacks. The new Setup() and IConfiguration support is indeed most welcome. 👍

**Comment by snakefoot** [comment]
Right now [ActivityTraceLayoutRenderer](https://github.com/NLog/NLog.DiagnosticSource/blob/54a06c0c06519cc15bd137e13ba8027ec58a3669/src/NLog.DiagnosticSource/LayoutRenderers/ActivityTraceLayoutRenderer.cs#L14) is sealed.

I guess one could [re-register with bonus-alias](https://github.com/NLog/NLog/wiki/Register-your-custom-component) `"ark.activityid"`.

**Comment by snakefoot** [comment]
Resolved by 357a2d2
